src/koheesio/categorical_encoding.py

```diff
@@ -0,0 +1,30 @@
+# Import necessary libraries
+from sklearn.preprocessing import OneHotEncoder
+import pandas as pd
+from pydantic import BaseModel
+
+# Define the EncodingConfig class
+class EncodingConfig(BaseModel):
+    drop_first: bool = True  # Whether to drop the first dummy column
+
+# Define the categorical_encoding function
+def categorical_encoding(data, columns, config: EncodingConfig):
```

Comment:
Please adjust this to become a `Step` class to take full advantage of Koheesio's built-in type checking.

Arguably, this should become a Pandas version of the Transformation class we have in the Spark module.

Reply:
Adjusted to using the Step class